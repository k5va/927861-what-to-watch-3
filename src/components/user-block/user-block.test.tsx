import * as React from "react";
import * as renderer from "react-test-renderer";
import {UserBlock} from "@components";
import {Provider} from "react-redux";
import {NameSpace} from "@store";
import configureStore from "redux-mock-store";
import {AuthorizationStatus} from "@consts";

const mockStore = configureStore([]);
const store = mockStore({
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
    user: null
  }
});

it(`UserBlock should render correctly`, () => {
  const wrapper = renderer
    .create(
        <Provider store={store}>
          <UserBlock />
        </Provider>
    )
    .toJSON();
  expect(wrapper).toMatchSnapshot();
});
