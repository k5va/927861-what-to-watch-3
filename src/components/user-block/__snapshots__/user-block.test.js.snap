// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock should render correctly 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="#"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
